bundles/core/src/main/java/com/adobe/cq/commerce/core/components/client/MagentoGraphqlClient.java

```diff
@@ -11,258 +11,36 @@
  *    governing permissions and limitations under the License.
  *
  ******************************************************************************/
-
 package com.adobe.cq.commerce.core.components.client;
 
-import java.time.OffsetDateTime;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.List;
 import java.util.Map;
-import java.util.Objects;
-import java.util.TimeZone;
-import java.util.stream.Collectors;
-
-import javax.servlet.http.Cookie;
 
-import org.apache.commons.lang3.StringUtils;
-import org.apache.http.Header;
-import org.apache.http.message.BasicHeader;
 import org.apache.sling.api.SlingHttpServletRequest;
 import org.apache.sling.api.resource.Resource;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.osgi.annotation.versioning.ProviderType;
 
-import com.adobe.cq.commerce.core.components.services.ComponentsConfiguration;
-import com.adobe.cq.commerce.graphql.client.CachingStrategy;
-import com.adobe.cq.commerce.graphql.client.CachingStrategy.DataFetchingPolicy;
+import com.adobe.cq.commerce.core.components.internal.client.MagentoGraphqlClientImpl;
 import com.adobe.cq.commerce.graphql.client.GraphqlClient;
 import com.adobe.cq.commerce.graphql.client.GraphqlClientConfiguration;
-import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
 import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
 import com.adobe.cq.commerce.graphql.client.HttpMethod;
-import com.adobe.cq.commerce.graphql.client.RequestOptions;
 import com.adobe.cq.commerce.magento.graphql.Query;
 import com.adobe.cq.commerce.magento.graphql.gson.Error;
-import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
-import com.adobe.cq.launches.api.Launch;
-import com.adobe.cq.wcm.launches.utils.LaunchUtils;
-import com.adobe.granite.ui.components.ds.ValueMapResource;
-import com.day.cq.commons.inherit.ComponentInheritanceValueMap;
-import com.day.cq.commons.inherit.HierarchyNodeInheritanceValueMap;
-import com.day.cq.commons.inherit.InheritanceValueMap;
-import com.day.cq.wcm.api.Page;
-import com.day.cq.wcm.api.PageManager;
 
 /**
- * This is a wrapper class for {@link GraphqlClient}. The constructor adapts a {@link Resource} to
- * the GraphqlClient class and also looks for the <code>magentoStore</code> property on the resource
- * path in order to set the Magento <code>Store</code> HTTP header. This wrapper also sets the custom
- * Magento Gson deserializer from {@link QueryDeserializer}.
+ * This interface gives access to a {@link GraphqlClient} configured for a given context.
+ * <p>
+ * It is implemented as adapter for {@link SlingHttpServletRequest} and {@link Resource} where the full feature set can only be used with
+ * a {@link SlingHttpServletRequest} as an adaptable.
+ *
+ * @see MagentoGraphqlClientImpl for what feature are supported for any of the adaptables
```

Comment:
Ideally the interface should be expressive enough to tell what you can expect without looking at the implementation and referring the implementation here could be avoided.

Reply:
True, but with Sling Models a lot of things are implicit and I am not sure how to express that adapting from a `Resource` provides only a subset of features than adapting from `SlingHttpServletRequest`.

I think as this is oss and in the same page it is good enough for a user to understand the difference.

Reply:
You can simply add it to the Javadoc of the interface.